### PR TITLE
Separate More Complete Media Sample

### PR DESCRIPTION
**Pull request by yschimke**
TODO - work out a sensible way to land. This works for my current testing on a branch.

#### WHAT

Split out the media sample into a distinct project.

#### WHY

It's way bigger than the rest of the samples. And we want to make it even bigger.

#### HOW

Delete and move classes, add enough to media3-backend to support real functionality.

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

**Review by luizgrp — COMMENTED** [review]
fyi this is not a full review, given the amount of changes I will be submitting more than one review

File: media-sample/src/main/java/com/google/android/horologist/mediasample/di/NetworkContainer.kt [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.LoggingEventListener
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+/**
+ * Simple DI implementation - to be replaced by hilt.
+ */
+class NetworkContainer(
```
[the remark this answers is not in the document]
Comment (yschimke) [reply media-sample/src/main/java/com/google/android/horologist/mediasample/di/NetworkContainer.kt:49]: I fear this is how DI libraries are born :)

Maybe should just use Dagger, or Koin.  Sort of shows why DI is useful.

File: media-sample/src/main/java/com/google/android/horologist/mediasample/ui/MediaPlayerScreenViewModel.kt [hunk 2]
```diff
@@ -14,31 +14,40 @@
...
+            while (isActive) {
+                delay(1000)
+                playerRepository.updatePosition()
+            }
+        }
     }
 
     class Factory(
-        private val playerRepositoryFactory: PlayerRepositoryImpl.Factory
+        private val playerRepositoryFactory: PlayerRepositoryImpl
```
Comment (luizgrp) [inline media-sample/src/main/java/com/google/android/horologist/mediasample/ui/MediaPlayerScreenViewModel.kt:44]: nit: rename to `playerRepository`

File: media3-backend/src/main/java/com/google/android/horologist/media3/player/PlayerRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+/**
+ * Repository for the current Media3 Player for a Player Activity.
+ *
+ * The current implementation is available as soon as the ListenableFuture
+ * to connect to the MediaSession completes.
+ */
+public class PlayerRepositoryImpl : PlayerRepository, Closeable {
```
Comment (luizgrp) [inline media3-backend/src/main/java/com/google/android/horologist/media3/player/PlayerRepositoryImpl.kt:43]: I thought this class would be provided in `media-data` module, which is empty at the moment. 
Unless you see `media-data` as a "utility" module with non-specific implementation?

File: media3-backend/src/main/java/com/google/android/horologist/media3/service/LifecycleMediaLibraryService.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import androidx.annotation.CallSuper
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ServiceLifecycleDispatcher
+import androidx.media3.session.MediaLibraryService
+import androidx.media3.session.MediaSession
+
+public abstract class LifecycleMediaLibraryService : MediaLibraryService(), LifecycleOwner {
+    @Suppress("LeakingThis")
+    private val mDispatcher = ServiceLifecycleDispatcher(this)
```
Comment (luizgrp) [inline media3-backend/src/main/java/com/google/android/horologist/media3/service/LifecycleMediaLibraryService.kt:30]: nit: drop Hungarian notation?

File: sample/src/main/java/com/google/android/horologist/sample/menuChips.kt [hunk 5]
```diff
@@ -193,11 +192,3 @@ fun FadeAwayChip(
         }
     )
 }
-
-@Composable
-fun MediaPlayerChip(navigateToRoute: () -> Unit) {
-    SampleChip(
-        onClick = { navigateToRoute() },
-        label = stringResource(id = R.string.chip_media_player_sample)
```
Comment (luizgrp) [inline sample/src/main/java/com/google/android/horologist/sample/menuChips.kt:201]: nit: remove `chip_media_player_sample` from `sample/../strings.xml`

File: tiles/src/main/java/com/google/android/horologist/tiles/CoroutinesTileService.kt [hunk 6]
```diff
@@ -69,7 +69,7 @@ public abstract class CoroutinesTileService : TileService(), LifecycleOwner {
      * See [onTileRequest] for most details.
      *
      * This runs a suspending function inside the lifecycleScope
-     * of the service on the Main thread.
+     * of the service on the Main tplaybackServicehrPlaybackServiceead.
```
Comment (luizgrp) [inline tiles/src/main/java/com/google/android/horologist/tiles/CoroutinesTileService.kt:72]: nit: something off here :D

File: media3-backend/src/main/java/com/google/android/horologist/media3/player/PlayerRepositoryImpl.kt (hunk 3 above)
Comment (yschimke) [reply media3-backend/src/main/java/com/google/android/horologist/media3/player/PlayerRepositoryImpl.kt:43]: Yep, happy to move it.

**Pull request closed without merging**